shared_model/validators/field_validator.cpp

```diff
@@ -208,7 +211,20 @@ namespace shared_model {
     void FieldValidator::validatePrecision(
         ReasonsGroupType &reason,
         const interface::types::PrecisionType &precision) const {
-      // define precision constraints
+      /* The following validation is pointless since PrecisionType is already
+       * uint8_t, but it is going to be changed and the validation will become
+       * meaningful.
+       */
+      auto min = std::numeric_limits<uint8_t>::min();
```

Comment:
`auto` will not work here because of the following signature `static constexpr T min();`, same with `max`. The `precision < min or precision > max` will probably perform incorrect cast.

Please check the following page and confirm if there will be no issues if `PrecisionType` will be unsigned, or have greater size than `uint8_t`. http://en.cppreference.com/w/cpp/language/operator_comparison